service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java

```diff
@@ -0,0 +1,153 @@
+
+        final Optional<Attachment> contentAttachment = documentReference.getContent().stream()
+            .map(DocumentReference.DocumentReferenceContentComponent::getAttachment)
+            .findFirst();
+
+        if (contentAttachment.isPresent() && contentAttachment.get().hasTitle()) {
+
+            final Attachment attachment = contentAttachment.get();
+            final String attachmentTitle = attachment.getTitle();
+            final String attachmentContentType = attachment.hasContentType()
+                ? attachment.getContentType() : DEFAULT_ATTACHMENT_CONTENT_TYPE;
+
+            builder.narrativeStatementId(narrativeStatementId)
+                .hasReference(true)
+                .comment(getComment(documentReference, attachmentTitle))
+                .availabilityTime(getAvailabilityTime(documentReference))
+                .referenceTitle(attachmentTitle)
+                .referenceContentType(attachmentContentType);
+        } else {
+            builder.narrativeStatementId(narrativeStatementId)
+                .comment(getComment(documentReference, null))
+                .availabilityTime(getAvailabilityTime(documentReference));
+        }
+
+        return TemplateUtils.fillTemplate(NARRATIVE_STATEMENT_TEMPLATE, builder.build());
+    }
+
+    private String getComment(final DocumentReference documentReference, final String attachmentTitle) {
+        StringBuilder commentBuilder = new StringBuilder();
+
+        mapType(documentReference, commentBuilder);
+        mapAuthorOrg(documentReference, commentBuilder);
+        mapCustodianOrg(documentReference, commentBuilder);
+        mapDescription(documentReference, commentBuilder);
+
+        Optional.ofNullable(attachmentTitle)
+            .ifPresent(title -> commentBuilder.append("Absent Attachment: ").append(title).append(StringUtils.SPACE));
+
+        mapSettings(documentReference, commentBuilder);
+
+        return commentBuilder.toString().trim();
+    }
+
+    private void mapSettings(final DocumentReference documentReference, final StringBuilder commentBuilder) {
+        final Optional<CodeableConcept> codeableConcept = Optional.ofNullable(documentReference.getContext())
+            .map(DocumentReference.DocumentReferenceContextComponent::getPracticeSetting);
+
+        codeableConcept
+            .filter(CodeableConcept::hasText)
+            .map(CodeableConcept::getText)
+            .ifPresentOrElse(text -> commentBuilder.append("Setting: ").append(text).append(StringUtils.SPACE),
+                () -> codeableConcept
+                    .filter(CodeableConcept::hasCoding)
+                    .map(CodeableConcept::getCoding)
+                    .flatMap(coding -> coding.stream().findFirst())
+                    .filter(Coding::hasDisplay)
+                    .map(Coding::getDisplay)
+                    .map(display -> commentBuilder.append("Setting: ").append(display).append(StringUtils.SPACE)));
```

Comment:
Just for the sake of using things the way they're meant, even if this already works:
```suggestion
                    .ifPresent(display -> commentBuilder.append("Setting: ").append(display).append(StringUtils.SPACE)));
```